### DISCUSSION TITLE
Concatenating Column and Table Name in compiles

### DISCUSSION THREAD

**Discussion by lukesmurray** [sqlalchemy/sqlalchemy#6828] [Usage Questions] [answered]
I would like to create a custom `ColumnClause` which concatenates the `tablename` with the value of a column.

I'm using this to generate unique ids with the structure `table_name/id`

As an example `select(TypedItemIdColumn(MyTable.id))` should yield the sql `SELECT 'MyTable' || '/' || id FROM MyTable`

I've gotten to the point that it almost works. It will output the correct sql if I select multiple columns but if I just select a `TypedItemIdColumn` by itself the `FROM` statement is not created by the compiler.

I do realize this is a bit hacky so any tips on a better way to accomplish what I'm going for would be much appreciated!

```python
class TypedItemIdColumn(ColumnClause):
    def __init__(self, column):
        self.tablename = column.class_.__tablename__
        self.column = column


@compiles(TypedItemIdColumn)
def compile_typeditemidcolumn(element, compiler, **kw):
    return compiler.process(f"{element.tablename}/" + element.column, **kw)
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

You should probably call `super().__init__` in the constructor to set the `_selectable` argument of it:
https://github.com/sqlalchemy/sqlalchemy/blob/84be0b42e03b0aaf8000637c4b5e348a39333e2f/lib/sqlalchemy/sql/elements.py#L4679

**Reply by lukesmurray** [reply]
Thank you!

**Reply by lukesmurray** [reply]
@CaselIT do you have a recommended way of getting the `Table` from a `Column` element. 

I'm passing in `MyTable.column` and I'm using the hacky `column._class.__tablename__` to get the table name but I'm not sure how to access the table itself to pass as the `_selectable`

**Reply by CaselIT** [reply]
I believe a Column has a `table` attribute
